services/softlayer_dns_domain_record.go

```diff
@@ -0,0 +1,139 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	datatypes "github.com/maximilien/softlayer-go/data_types"
+	softlayer "github.com/maximilien/softlayer-go/softlayer"
+)
+
+type softLayer_Dns_Domain_Record_Service struct {
+	client softlayer.Client
+}
+
+func NewSoftLayer_Dns_Domain_Record_Service(client softlayer.Client) *softLayer_Dns_Domain_Record_Service {
+	return &softLayer_Dns_Domain_Record_Service{
+		client: client,
+	}
+}
+
+func (sldr *softLayer_Dns_Domain_Record_Service) GetName() string {
+	return "SoftLayer_Dns_Domain_ResourceRecord"
+}
+
+// Provides SoftLayer resource type by record type
+func (sldr *softLayer_Dns_Domain_Record_Service) getNameByType(dnsType string) string  {
```

Comment:
move down to `//Private methods` section. Also, no need to have comment outside of the method as it distracts. The comment inside is OK since it makes clear the types. Ideally, a link to SL documentation would be best.